src/compiler/checker.ts

```diff
@@ -20220,7 +20225,7 @@ namespace ts {
             checkDecorators(node);
         }
 
-        function checkTypeArgumentConstraints(typeParameters: TypeParameter[], typeArgumentNodes: ReadonlyArray<TypeNode>): boolean {
+        function checkTypeArgumentConstraints(typeParameters: TypeParameter[], typeArgumentNodes: ReadonlyArray<TypeNode>, constraints?: (Type | undefined)[]): boolean {

```

Comment:
Some kind of comment indicating that `constraints` is an output array (or a more helpful name) would be useful. It's not immediately obvious what the argument is for.

Reply:
This probably also needs a `reportErrors` boolean parameter to control weatehr or not the node is given to `checkTypeAssignableTo` (so `getTypeArgumentConstraint` can set it to `false`) - as @DanielRosenwasser said, the version of this in the checker API shouldn't be able to report a diagnostic. Plus, potentially generating a diagnostic while generation completions would be odd.